docs/jwst/user_documentation/running_pipeline_python.rst

```diff
@@ -529,3 +529,196 @@ individual step parameter must be set when using this method, or else the coded
+set to `quarter`, `half`, `all`, or `none`, which is the default value.
+
+The following example turns on the step multiprocessing while setting up a log
+file for each run of the pipeline and a text file with the full traceback in case
+there is a crash. Notice only one of the steps has multiprocessing turned on. We
+do not recommend to simultaneously enable both steps to do multiprocessing, as
+this may likely lead to running out of system memory.
+
+
+
+::
+
+    # SampleScript1
+
+    import os, sys
+    import traceback
+    import configparser
+    from glob import glob
+    from jwst.pipeline import Detector1Pipeline
+
+    files = glob('*uncal.fits')
+    output_dir = '/my_project'
+
+    def mk_stpipe_log_cfg(output_dir, log_name):
+        """
+        Create a configuration file with the name log_name, where
+        the pipeline will write all output.
+        Args:
+            outpur_dir: str, path of the output directory
+            log_name: str, name of the log to record screen output
+        Returns:
+            nothing
+        """
+        config = configparser.ConfigParser()
+        config.add_section("*")
+        config.set("*", "handler", "file:" + log_name)
+        config.set("*", "level", "INFO")
+        pipe_log_config = os.path.join(output_dir, "pipeline-log.cfg")
+        config.write(open(pipe_log_config, "w"))
+
+    def main():
+        for item in files:
+            fle = os.path.basename(item).replace('.fits', '')
+            log_name = os.path.join(output_dir, fle)
+            mk_stpipe_log_cfg(output_dir, log_name+'.log')
+            det1 = Detector1Pipeline()
+            parameter_dict = {"ramp_fit": {"maximum_cores": 'all'}}
+            pipe_success = False
+            print('Running Detector 1 on file: ', item)
+            try:
+                det1.call(item, save_results=True, steps=parameter_dict, output_dir=output_dir, logcfg="pipeline-log.cfg")
+                pipe_success = True
+                print('\n * Pipeline finished for file: ', item, ' \n')
+            except Exception:
+                print('\n *** OH NO! The detector1 pipeline crashed! *** \n')
+                pipe_crash_msg = traceback.print_exc()
+            if not pipe_success:
+                crashfile = open(log_name+'_pipecrash.txt', 'w')
+                print('printing file with crash  message')
+                print(pipe_crash_msg, file=crashfile)
+
+        print('\n * Finished multiprocessing! \n')
+
+    if __name__ == "__main__":
+        main()
+
+
+2. Calling the pipeline using multiprocessing. The following example uses this
+option setting up a log file for each run of the pipeline and a text file with
+the full traceback in case there is a crash. Notice that the ``import`` statement
+of the pipeline is within the multiprocessing block that gets called by every
+worker. This is to avoid a known memory leak.
+
+
+::
+
+    # SampleScript2
```

Comment:
This one looks great.  In contrast with the first example script, it makes sense to set up the logfile with a script here because a separate one is being created for each filename.